phoenix-core/src/it/java/org/apache/phoenix/end2end/MutationBatchFailedStateMetricIT.java

```diff
@@ -0,0 +1,146 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@Category(ParallelStatsDisabledTest.class)
+@RunWith(Parameterized.class)
+public class MutationBatchFailedStateMetricIT extends ParallelStatsDisabledIT {
+
+    String create_table =
+            "CREATE TABLE IF NOT EXISTS %s(ID VARCHAR NOT NULL PRIMARY KEY, VAL1 INTEGER, VAL2 INTEGER)";
+    String indexName = generateUniqueName();
+    String create_index = "CREATE INDEX " + indexName + " ON %s(VAL1 DESC) INCLUDE (VAL2)";
+    String upsertStatement = "UPSERT INTO %s VALUES(?, ?, ?)";
+    String deleteTableName = generateUniqueName();
+    private final boolean transactional;
+
+    public MutationBatchFailedStateMetricIT(String transactionProvider) {
+        this.transactional = transactionProvider != null;
+        if (this.transactional) {
+            create_table =
+                    create_table + (this.transactional
+                            ? (" TRANSACTIONAL=true,TRANSACTION_PROVIDER='" + transactionProvider
+                                    + "'")
+                            : "");
+            System.out.println(create_table);
+        }
+        createTables();
+        populateTables();
+    }
+
+    @BeforeClass
+    public static synchronized void doSetup() throws Exception {
+        Map<String, String> serverProps = Maps.newHashMapWithExpectedSize(3);
+        serverProps.put("hbase.coprocessor.abortonerror", "false");
+        serverProps.put(Indexer.CHECK_VERSION_CONF_KEY, "false");
+        Map<String, String> clientProps = Maps.newHashMapWithExpectedSize(2);
+        clientProps.put(QueryServices.COLLECT_REQUEST_LEVEL_METRICS, String.valueOf(true));
+        clientProps.put(QueryServices.TRANSACTIONS_ENABLED, "true");
+        setUpTestDriver(new ReadOnlyProps(serverProps.entrySet().iterator()),
+            new ReadOnlyProps(clientProps.entrySet().iterator()));
+
+    }
+
+    @Parameterized.Parameters(name = "MutationBatchFailedStateMetricIT_transactionProvider={0}")
+    public static synchronized Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] { { "OMID" }, { null } });
+    }
+
+    @Test
+    public void testFailedDelete() throws SQLException {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        Connection conn = null;
+        try {
+            conn = DriverManager.getConnection(getUrl(), props);
+            long rowCount = TestUtil.getRowCount(conn, deleteTableName);
+            System.out.println("Row Count:" + rowCount);
+
+            Connection finalConn = conn;
+            boolean autoCommit = conn.getAutoCommit();
+            System.out.println("Auto commit : " + autoCommit);
+            Statement stmt = conn.createStatement();
+
+            TestUtil.addCoprocessor(conn, deleteTableName,
+                ImmutableIndexIT.DeleteFailingRegionObserver.class);
+
+            String dml = String.format("DELETE FROM %s where val1 >  1", deleteTableName);
```

Comment:
Adding a comment in the code helps. Something like this will delete 4 out of 5 rows from the table.